### DISCUSSION TITLE
What are your must have features?  Here's your chance to let us know what we should work on

### DISCUSSION THREAD

**Discussion by dinki** [dinki/View-Assist#19] [Q&A]
I started View Assist as a possible replacement for my Alexa devices.  While it was a lot of work to get here, I'm within striking distance of being able to do most of what we want.  I'm posting this to see what YOU want to do with your voice assistant.  To kick off the discussion, I would like to list what I want and will tag the things that are already in place with an asterisk

* Tell me the time *
* Tell me the weather *
* Tell me if it will rain today or tomorrow *
* Play Spotify (Music Assistant) *
* Play Tunein Radio stations (Music Assistant) *
* Make broadcast announcements to all speakers *
* Do general knowledge search (limited through Wikipedia) *
* Drop in to another satellite device(s) and have a conversation like an intercom
* Show security cameras *
* Add/remove items from shopping list (add works but not remove) half *
* Announce when washer/dryer are done, alarm is set, etc *
* Resume music playback after announcements
* Ability to put device in do not disturb mode *
* Show traffic conditions and travel time for commute *
* Set thermostat by saying I'm cold/I'm hot
* Show recipes (Mealie) and any other website by voice *
* Set alarm on individual devices (working but no snooze) half *
* Set reminder on individual devices (working but no display yet) half *
* Tell me a joke *
* Respond when I say thank you *
* Dim screen for sleeping but make clock still visible *
* Some sort of visual notice when I am on a Teams call/meeting so family can not be their normal selves
* Show sports scores when asked
* How do you say 'word or phrase' in Spanish/German/French/etc
* How do you spell 'word'
* Basic math
* Repeat what you just said
* SHOW ME THE CLOCK INSTEAD OF ADS *********************************************

This is just what I can think of now.  I will be releasing all of these soon and want to start working on the other things that you guys would like to use.

**Comment by mngarchow** [comment]
I'd like to be able to dictate messages to be sent to HA mobile devices (the companion app can receive notifications from your local HA even if you don't have HA Cloud)
I'd like to be able to talk to View Assist in multiple languages (I am working on figuring that out now).

**Reply by dinki** [reply]
> I'd like to be able to dictate messages to be sent to HA mobile devices (the companion app can receive notifications from your local HA even if you don't have HA Cloud)

Can you tell me how you would do this?  'Ok Nabu dictate a message to devicename' ?  HA dev team is working on ability to do 'two way conversations' where it can listen to your response.  Once that would be in place it could say 'What's the message' and then you could dictate.  Not sure how else you could do this short of doing something like 'Ok Nabu tell devicename Here is my message'

**Reply by mngarchow** [reply]
I'd probably do this one: 'Ok Nabu tell devicename Here is my message'

"Ok Nabu, tell scratt's phone Dinner will be ready at 7pm".. then HA sends the notification via firebase to my mobile device which pops up the notification (I already have this set up for garage door open notifications, does not require HA Cloud subscription). It may not require view assist at all, but the VA device would be the microphone I would use. 

I only thought about it because of your item broadcast announcements to other VA devices.

**Comment by jlmaners** [comment]
One thing we love about our Google Nest Hubs is the ability to have it show photos and rotate through them. This shows a mini clock and weather display as part of it - this can likely be handled via a custom dashboard/card combos. 

I'd encourage that the default dashboard or home dashboard shown not be targeted at one specific use case ie only a clock.

Things we make voice calls for on our Google devices and expect a display on a near daily basis:
- what's the weather like today?
- Add to shopping list
- Create temporary, named timers (know some of this is in the works in HA which would be a prerequisite)
- What's traffic look like on way to work?
- general information inquiries
- unit conversions & math functions
- play music
- broadcast to all devices
- broadcast to targeted device

**Reply by dinki** [reply]
Great information and great news on most of these.

I have been thinking about the rotating photos.  Please share a screenshot of what you have (if you can or something generic).

The request for the default home screen to be configurable is already in place.  This is controlled by a control automation and is set on a per device basis so you choose any of the VA views as home or another view from another dashboard if you choose.

All items you have listed with the exception of 'unit conversions and math functions' are available on my system now.  A couple of notes though:

* shopping list does not allow for removal but does add.  I know how to fix it I just haven't had the time
*  General information inquiries are done using wikipedia (still a work in progress).  
* Broadcast to all devices will broadcast to all devices in the View Assist satellite group that are not in do not disturb mode
* Broadcast to a single device not in place but is easily adapted from the all devices automation
* Disposable timers are now part of HA core BUT no information on using them has been released yet (coming this month though)

I think you are on the Discord server, right?  I can provide you with early access (aka beta testing) on the control automation and the broadcast to see how it works.  Do you have multiple VA devices configured?

**Reply by psgcooldog** [reply]
Yes, definitely Timers.  I need to research how to create Alarms.  Is that in the released version?

**Reply by autumnwalker** [reply]
re: photos - I have my View Assist satellite connected to Immich Kiosk using guidance from this wiki entry: https://dinki.github.io/View-Assist/docs/community-contributions/cc-tipsandtricks/immich_clock_background

It works really well!

**Comment by elwing00** [comment]
My kid likes to use her echo for homework (this is more of an echo has it feature, that I really don't mind if she doesn't have...):

- What is 3x4?
- How do you spell X

**Comment by elwing00** [comment]
Notification of "something" - 
"OK Nabu, what's the notification?" then it reads off the notification (or displays it)

**Comment by atomicpapa** [comment]
Different backgrounds on different devices

**Comment by euroman2007** [comment]
is there a way to let the screen of the Lenovo Thinksmart dim and brighten automaticly with the lightsensor that is build into the Lenovo Thinksmart. so you wont have to ask it to go to nightmode every night ?
i wold love this function. then all of my devices use a little less power overnight. and the one next to my bed wont keep me awake :P

**Comment by autumnwalker** [comment]
Add notification for weather alerts. My weather integration (Environment Canada) provides a sensor.warnings which has attributes for number of active warnings, text of the warning (e.g. "rainfall warning") and location name.

The way Echo Show devices handle this is by showing a conditional view with "<warning description> for <location>". You can also ask it what the warning is and it will read back the details to you.

I've mocked up a new clock view locally where it also will conditionally display a weather warning line item that when tapped (currently) will show the weather card. I'm thinking about creating the Echo Show card above and having it link to that instead.

**Comment by autumnwalker** [comment]
Ability to choose which views you want in the cycle mode.

Ideally only views which are in use / relevant get cycled versus today all views (e.g. music view) are included in the cycle.

**Comment by euroman2007** [comment]
i have purchased 6 lenovo smart view displays right after i had seen your project a while ago. the only thing i still love to have is that i can use a view in landscape mode and a view in portrait mode. i cant figure out how to do that. i hope you are willing to  help me with that :D  Greetings Fernando

**Reply by jpbaril** [reply]
I don't think it is possible to have VA in portrait mode right now. But there seems to be a PR in progress for that feature: https://github.com/dinki/View-Assist/pull/224

**Comment by DhananjayRamachandran** [comment]
Awesome work creating a great dashboard view!

One request for a feature I’d love to have included is to be able to customize the buttons shown in the intent view to include a slider if necessary.

For example when requesting to turn on a specific scene involving 3 lights (only 1 of which has brightness control) there would be 2 standard button cards and 1 slider button card for the appropriate entity (perhaps using [mattieha/slider-button-card](https://github.com/mattieha/slider-button-card)) to control the brightness of the one light. 

If this isn’t easily doable, would it possible to show the “more info” panel instead of a button card within the layout view?

**Comment by omersiar** [comment]
Implement android app with ESPHome protocol for broader audience and easier integration and reducing the onboarding requirements, it would be then exposed as ESPHome device and easily added to voice pipeline

Python Example
https://github.com/OHF-Voice/linux-voice-assistant

Java
https://medium.com/@pachdjian/esphome-and-protocoll-buffer-530cd9d2598b
https://github.com/paoloach/LuminousSaucer/tree/main/app/SaucerLuminous/app/src/main/java/it/achdjian/saucer/luminous

**Comment by mcmasterp** [comment]
Maybe I'm just missing how to use this feature but I'd love a night mode where the background is black, the clock is red, and the screen is very dim

**Comment by mandebooks** [comment]
ESP32-S3 Device with Display as a View Assist Device - not just a voice device, but full display access.

**Comment by mcmasterp** [comment]
Smaller clock for devices like rooted echo spot

**Comment by bobobado** [comment]
I'd love it if I could call an unsplash URL similar to wallpanel so I can dictate my background images rather than random only. Thank you for the incredible work!

https://unsplash.com/documentation#get-a-random-photo

**Comment by Judman** [comment]
Just found this project and all the amazing things others have done with it. However, I have no desire to use voice. I don't like talking to my electronics so being able to use this completely without voice would be amazing. Maybe not using voice is the root cause of my issue below, but thats my first request, the clear option to set up and use completely without voice.

Second is I have not been able to wrap my brain around customization. I have the homepage displaying on my jail broken echo show 5, but that's as far as I seem to be able to get. I can't seem to use the camera's or timers or any of the other dashboards. Again, I am dense and probably doing something wrong, but the doc's don't really clearly spell out what is needed to fully utilize view assistant as is initially installed.

**Comment by judebert** [comment]
I have the same goal: completely leave the Amazon system for a fully HA solution. I'd prefer to do all my processing in-house, although Nabu Casa is a reasonable second choice.

I surveyed my entire family about how they use their Echos. This is our prioritized list:
1. Local weather
2. Grocery lists (very difficult with StP)
3. Timers
  * Multiple, named, timers (hard but feasible with StP)
  * Alarms
4. Music (Apple Music and/or local library)
  * Playlists (easily feasible with StP)
  * Arbitrary songs (very difficult, possibly infeasible with StP)
  * Volume control ("louder", "quieter by 3", "set volume 4")
5. Smart device control
  * On/off
  * Brightness
  * Color
6. Arbitrary weather (hard but feasible with StP when limited)
7. Weather notifications
8. General queries
  * Unit conversions
  * Upcoming media releases
  * Biographies
9. Intercom

If we can complete the first 5, we can ditch our Echoes. Everything after that just makes us more satisfied with leaving Amazon.

**Comment by NickTheZonie** [comment]
I would love to be able to ask the voice assistant to play an album as opposed to a song/playlist or artist!
